providers/base/bin/prime_offload_tester.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python3
+"""prime offload test module."""
+# This file is part of Checkbox.
+#
+# Copyright 2023 Canonical Ltd.
+# Written by:
+#   Hanhsuan Lee <hanhsuan.lee@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import threading
+import subprocess
+import time
+import re
+import json
+import argparse
+import logging
+from enum import IntEnum
+
+
+class PrimeOffloaderError(IntEnum):
```

Comment:
This is not how error handling is done in python.
Here's a good resource on it: https://docs.python.org/3/tutorial/errors.html